File: frontend/src/data/workspaces.ts
import type { WorkspaceModule } from "@/app/auth";

export type SurveyWorkspace = {
  slug: WorkspaceModule;
  label: string;
  shortLabel: string;
  formId: string;
  schema: string;
  description: string;
  accent: string;
};

export const SURVEY_WORKSPACES: SurveyWorkspace[] = [
  {
    slug: "spread",
    label: "Spread Category",
    shortLabel: "Spread",
    formId: "BHT_3_Category_Survey_Spread_Wave_1",
    schema: "spread",
    description: "Margarine and spreads Wave 1 quality-control workspace.",
    accent: "from-amber-400 to-orange-500",
  },
  {
    slug: "edible-oil",
    label: "Edible Oil Category",
    shortLabel: "Edible Oil",
    formId: "BHT_3_Category_Survey_Edible_oil_Wave_1",
    schema: "edible_oil",
    description: "Edible oil Wave 1 quality-control workspace.",
    accent: "from-emerald-400 to-teal-600",
  },
  {
    slug: "breakfast-cereal",
    label: "Breakfast Cereal Category",
    shortLabel: "Breakfast Cereal",
    formId: "BHT_3_Category_Survey_Breakfast_wave_1",
    schema: "breakfast_cereal",
    description: "Breakfast cereal Wave 1 quality-control workspace.",
    accent: "from-sky-400 to-blue-600",
  },
];

export function getSurveyWorkspace(slug: string | null | undefined) {
  return SURVEY_WORKSPACES.find((workspace) => workspace.slug === slug) ?? null;
}

File: src/survey_platform/workspaces.py
from __future__ import annotations

from dataclasses import dataclass
from pathlib import Path

from survey_platform.config import load_dotenv_file, read_env


@dataclass(frozen=True, slots=True)
class SurveyWorkspace:
    slug: str
    label: str
    schema: str
    form_id: str
    dictionary_file: Path


WORKSPACE_DEFINITIONS = (
    ("spread", "Spread Category", "spread", "BHT_3_Category_Survey_Spread_Wave_1", "BHT_3_Categories_Margarine_Wave_1_Updated_Script.xlsx"),
    ("edible-oil", "Edible Oil Category", "edible_oil", "BHT_3_Category_Survey_Edible_oil_Wave_1", "BHT_3_Categories_Edible_Oil_Wave_1_Updated_Script.xlsx"),
    ("breakfast-cereal", "Breakfast Cereal Category", "breakfast_cereal", "BHT_3_Category_Survey_Breakfast_wave_1", "BHT_3_Categories_Breakfast_Cereal_Wave_1_Updated_Script.xlsx"),
)


def load_survey_workspaces(base_dir: Path | None = None) -> list[SurveyWorkspace]:
    root = (base_dir or Path.cwd()).resolve()
    dotenv = load_dotenv_file(root / ".env")
    result: list[SurveyWorkspace] = []
    for slug, label, schema, default_form_id, filename in WORKSPACE_DEFINITIONS:
        env_prefix = schema.upper()
        form_id = read_env(f"SURVEYCTO_{env_prefix}_FORM_ID", dotenv, default_form_id) or default_form_id
        dictionary = read_env(
            f"SURVEYCTO_{env_prefix}_XLSFORM",
            dotenv,
            f"data/category_xlsforms/{filename}",
        ) or f"data/category_xlsforms/{filename}"
        result.append(SurveyWorkspace(slug, label, schema, form_id, root / dictionary))
    return result
